perun-core/src/test/java/cz/metacentrum/perun/core/entry/GroupsManagerEntryIntegrationTest.java

```diff
@@ -3483,6 +3483,51 @@ public void addAndRemoveMemberInNonSynchronizedGroup() throws Exception {
 		assertTrue("List of members should be empty", members.isEmpty());
 	}
 
+	@Test
+	public void addMemberToGroups() throws Exception {
+		System.out.println(CLASS_NAME + "addMemberToGroups");
+
+		vo = setUpVo();
+
+		groupsManagerBl.createGroup(sess, vo, group);
+		groupsManagerBl.createGroup(sess, vo, group2);
+		groupsManagerBl.createGroup(sess, vo, group3);
+
+		List<Group> groups = Arrays.asList(group, group2, group3);
+
+		Attribute synchroAttr = new Attribute(perun.getAttributesManager().getAttributeDefinition(sess, GroupsManager.GROUPSYNCHROENABLED_ATTRNAME));
```

Comment:
I don't think you need to test "false" case - when synchronization is disabled. We do tests this logic in other tests.

Basically you should have tests for each case - group/member not from same vo, synchronized group, not synchronized group and lastly added logic about group structure synchronization.

But I think it is basically covered in other tests. Please check if there are tests for group structure synchronization case. This could be added for standard add/removeMember() tests.